### ISSUE TITLE
Cannot cast org.hl7.fhir.r4.model.CodeSystem to org.hl7.fhir.r4.model.StructureDefinition

### ISSUE THREAD

**Issue by larisapetrov** [gematik/app-referencevalidator#12]
**Steps to Reproduce:**
- Ensure Gematik Validator version **2.0.2** is installed
- Latest version of plugin used: isik2-basismodul-2.0.0
- Attempt to validate a Condition resource that contains a code.coding.extension (you can find the file we are using for validation: [test-ISIK_Condition-defect_validator.json](https://github.com/gematik/app-referencevalidator/files/14384655/test-ISIK_Condition-defect_validator.json))
- Observe the validator crash.
<img width="967" alt="image" src="https://github.com/gematik/app-referencevalidator/assets/134613069/f8f28f4a-3261-4f4e-954b-a2f87b0b2501">

**Expected Behavior:**
The Gematik Validator should successfully validate the Condition resource without crashing, similar to the behavior from version **2.0.1**.

**Affected Version: 2.0.2
Working Version: 2.0.1**

Thank you for your attention to this matter!

**Full stack trace:**
23.02.24-12:21:33.077 ERROR - Exception
java.lang.ClassCastException: Cannot cast org.hl7.fhir.r4.model.CodeSystem to org.hl7.fhir.r4.model.StructureDefinition
        at java.base/java.lang.Class.cast(Unknown Source)
        at ca.uhn.fhir.context.support.IValidationSupport.fetchResource(IValidationSupport.java:218)
        at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.fetchResource(ValidationSupportChain.java:252)
        at org.hl7.fhir.common.hapi.validation.validator.VersionSpecificWorkerContextWrapper.lambda$new$0(VersionSpecificWorkerContextWrapper.java:86)
        at com.github.benmanes.caffeine.cache.LocalLoadingCache.lambda$newMappingFunction$3(LocalLoadingCache.java:197)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$13(BoundedLocalCache.java:2550)
        at java.base/java.util.concurrent.ConcurrentHashMap.compute(Unknown Source)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2548)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2531)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:110)
        at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:58)
        at ca.uhn.fhir.sl.cache.caffeine.LoadingCacheDelegator.get(LoadingCacheDelegator.java:39)
        at org.hl7.fhir.common.hapi.validation.validator.VersionSpecificWorkerContextWrapper.fetchResource(VersionSpecificWorkerContextWrapper.java:352)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkExtension(InstanceValidator.java:1840)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChildByDefinition(InstanceValidator.java:5565)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChild(InstanceValidator.java:5393)
        at org.hl7.fhir.validation.instance.InstanceValidator.validateElement(InstanceValidator.java:5341)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChildByDefinition(InstanceValidator.java:5670)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChild(InstanceValidator.java:5393)
        at org.hl7.fhir.validation.instance.InstanceValidator.validateElement(InstanceValidator.java:5341)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChildByDefinition(InstanceValidator.java:5670)
        at org.hl7.fhir.validation.instance.InstanceValidator.checkChild(InstanceValidator.java:5393)
        at org.hl7.fhir.validation.instance.InstanceValidator.validateElement(InstanceValidator.java:5341)
        at org.hl7.fhir.validation.instance.InstanceValidator.startInner(InstanceValidator.java:4993)
        at org.hl7.fhir.validation.instance.InstanceValidator.start(InstanceValidator.java:4786)
        at org.hl7.fhir.validation.instance.InstanceValidator.validateResource(InstanceValidator.java:6225)
        at org.hl7.fhir.validation.instance.InstanceValidator.validate(InstanceValidator.java:884)
        at org.hl7.fhir.validation.instance.InstanceValidator.validate(InstanceValidator.java:718)
        at org.hl7.fhir.common.hapi.validation.validator.ValidatorWrapper.validate(ValidatorWrapper.java:193)
        at org.hl7.fhir.common.hapi.validation.validator.FhirInstanceValidator.validate(FhirInstanceValidator.java:238)
        at org.hl7.fhir.common.hapi.validation.validator.BaseValidatorBridge.doValidate(BaseValidatorBridge.java:22)
        at org.hl7.fhir.common.hapi.validation.validator.BaseValidatorBridge.validateResource(BaseValidatorBridge.java:48)
        at org.hl7.fhir.common.hapi.validation.validator.FhirInstanceValidator.validateResource(FhirInstanceValidator.java:26)
        at ca.uhn.fhir.validation.FhirValidator.validateResource(FhirValidator.java:342)
        at ca.uhn.fhir.validation.FhirValidator.doValidate(FhirValidator.java:263)
        at ca.uhn.fhir.validation.FhirValidator.validateWithResult(FhirValidator.java:236)
        at de.gematik.refv.commons.validation.GenericValidator.validateUsingDependencyList(GenericValidator.java:211)
        at de.gematik.refv.commons.validation.GenericValidator.validateWithoutConfiguredLocator(GenericValidator.java:163)
        at de.gematik.refv.commons.validation.GenericValidator.validate(GenericValidator.java:113)
        at de.gematik.refv.commons.validation.BaseValidationModule.validateString(BaseValidationModule.java:71)
        at de.gematik.refv.commons.validation.BaseValidationModule.validateFile(BaseValidationModule.java:86)
        at de.gematik.refv.cli.ReferenceValidator.run(ReferenceValidator.java:126)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at de.gematik.refv.cli.ReferenceValidator.main(ReferenceValidator.java:99)

**Comment by alexey-tschudnowsky** [comment]
Hello @larisapetrov ,
thank you for posting the issue. We will analyze, why the validator version 2.0.1 didn't recognize any issues, but there is a problem with the instance - the extension _icd-10-gm-mehrfachcodierungs-kennzeichen_ is incorrectly used:

```
"extension": [
      {
          "url": "http://fhir.de/CodeSystem/icd-10-gm-mehrfachcodierungs-kennzeichen",
          "valueString": "*"
      }
  ]
```

The correct way would be:

```
"extension":  [
      {
          "url": "http://fhir.de/StructureDefinition/icd-10-gm-mehrfachcodierungs-kennzeichen",
          "valueCoding": {
              "version": "2021",
              "code": "*",
              "system": "http://fhir.de/CodeSystem/icd-10-gm-mehrfachcodierungs-kennzeichen"
          }
      }
  ]
```

Cf. example at https://simplifier.net/guide/isik-basis-v2/ImplementationGuide-markdown-Datenobjekte-Datenobjekte-Diagnose?version=current

Nevertheless, for such an instance one would expect a validation issue of error severity instead of a crash, so we will plan a fix for the next release.

**Comment by larisapetrov** [comment]
Hi @alexey-tschudnowsky 
Thank you for bringing this to our attention. You are absolutely right, the Condition is incorrect and we will take steps to fix it.
We missed the issue because, as mentioned, the previous version of the validator didn't catch this.

Thanks again!

**Comment by alexey-tschudnowsky** [comment]
Dear @larisapetrov ,
thanks again for the reporting the issue. It has been fixed with the [2.1.1 release](https://github.com/gematik/app-referencevalidator/releases/tag/2.1.1) of the validator. 
Best Regards,
Alexey
